src/Features/Core/Portable/Diagnostics/Analyzers/NamingStyles/EditorConfigNamingStyleParser.Dictionary.cs

```diff
@@ -0,0 +1,245 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.CodeAnalysis.Diagnostics.Analyzers.NamingStyles
+{
+    internal static partial class EditorConfigNamingStyleParser
+    {
+        public static IEnumerable<Tuple<string, string, Tuple<string, string, string, string>, Tuple<string, string, string, string, string>>> ParseDictionary(IReadOnlyDictionary<string, object> allRawConventions)
+        {
+            var namingRules = (from kvp in allRawConventions
+                               where kvp.Key.Trim().StartsWith("dotnet_naming_rule.")
+                               select kvp).ToDictionary(x => x.Key, x => x.Value);
+
+            var namingRulesSeverity = (from kvp in namingRules
+                                       where kvp.Key.Trim().EndsWith(".severity")
+                                       select kvp).ToDictionary(x => x.Key, x => x.Value);
+
+            var namingRulesSymbols = (from kvp in namingRules
+                                      where kvp.Key.Trim().EndsWith(".symbols")
+                                      select kvp).ToDictionary(x => x.Key, x => x.Value);
+
+            var namingRulesStyle = (from kvp in namingRules
+                                    where kvp.Key.Trim().EndsWith(".style")
+                                    select kvp).ToDictionary(x => x.Key, x => x.Value);
+
+            var namingSymbols = (from kvp in allRawConventions
+                                 where kvp.Key.Trim().StartsWith("dotnet_naming_symbols.")
+                                 select kvp).ToDictionary(x => x.Key, x => x.Value);
+
+            var namingSymbolsApplicableKinds = (from kvp in namingSymbols
+                                                where kvp.Key.Trim().EndsWith(".applicable_kinds")
+                                                select kvp).ToDictionary(x => x.Key, x => x.Value);
+
+            var namingSymbolsApplicableaccessibilities = (from kvp in namingSymbols
+                                                          where kvp.Key.Trim().EndsWith(".applicable_accessibilities")
+                                                          select kvp).ToDictionary(x => x.Key, x => x.Value);
+
+            var namingSymbolsRequiredModifiers = (from kvp in namingSymbols
+                                                  where kvp.Key.Trim().EndsWith(".required_modifiers")
+                                                  select kvp).ToDictionary(x => x.Key, x => x.Value);
+
+            var namingStyles = (from kvp in allRawConventions
+                                where kvp.Key.Trim().StartsWith("dotnet_naming_style.")
+                                select kvp).ToDictionary(x => x.Key, x => x.Value);
+
+            var namingStyleRequiredPrefix = (from kvp in namingStyles
```

Comment:
It would seem easier to just loop through all keys, do the split you're doing around line 63, and then just switch on the last component.